com.microsoft.mrtk.uxcore/Dialog/Dialog.cs

```diff
@@ -2,259 +2,245 @@
+        }
+        
+        /// <inheritdoc />
+        public virtual IDialog Show()
         {
-            GameObject dialogGameObject = Instantiate(dialogComponentOnPrefab.gameObject);
-
-            Dialog dialog = dialogGameObject.GetComponent<Dialog>();
-            if (property != null)
-            {
-                dialog.Property = property;
-                dialog.Property.TargetDialog = dialog;
-            }
+            headerText.gameObject.SetActive(header != null);
+            headerText.text = header;
+            bodyText.gameObject.SetActive(body != null);
+            bodyText.text = body;
 
-            dialog.PlaceForNearInteraction = placeForNearInteraction;
+            positiveButton.Interactable.gameObject.SetActive(positiveAction != null);
+            negativeButton.Interactable.gameObject.SetActive(negativeAction != null);
+            neutralButton.Interactable.gameObject.SetActive(neutralAction != null);
 
-            if (openOnInstantiate)
-            {
-                if (property == null)
-                {
-                    Debug.LogError("Cannot open the dialog because the property is not specified");
-                    dialogGameObject.SetActive(false);
-                    return dialog;
-                }
-                dialog.Open();
-            }
-            else
-            {
-                dialogGameObject.SetActive(false);
-            }
+            gameObject.SetActive(true);
 
-            return dialog;
+            return this;
         }
 
-        protected virtual void Awake()
+        /// <inheritdoc />
+        public virtual void Dismiss()
         {
-            SetInteractionMode(PlaceForNearInteraction);
+            negativeAction?.RemoveAllListeners();
+            positiveAction?.RemoveAllListeners();
+            neutralAction?.RemoveAllListeners();
+            
+            // Only invoke the Dismissed callback after we've played the full dismissal animation.
+            StartCoroutine(InvokeDismissalAfterAnimation());
         }
 
+        /// <inheritdoc />
+        public GameObject VisibleRoot => gameObject;
+
         /// <summary>
-        /// Apply settings to the follow solver based on the selected interaction mode.
-        /// </summary>
-        [Obsolete("Dialog's ConstantViewSize/variable follow distance is deprecated until Dialog is refactored to render correctly when scaled.")]
-        protected virtual void SetInteractionMode(bool useNearInteractionPlacement) { }
+        /// Coroutine to set the animation trigger, wait for the animation to finish,
+        /// and then hide the dialog and invoke the dismissal action. This coroutine
+        /// is started by the base <see cref="Dismiss"/> method once all listeners
+        /// have been removed.
+        private IEnumerator InvokeDismissalAfterAnimation()
+        {
```

Comment:
Might want to put some sort of locking mechanism on this co-routine so repeated clicks don't result in weird dismissal animation behaviour.

Reply:
Especially since onDIsmissed's timing is dependent on the animation completing.

Reply:
It already "locks" by immediately removing the button listeners whenever an option is selected. You can "click" the buttons multiple times, but nothing happens. The moment one button is clicked, the "deed is done" and nothing is responsive after that.

(you can test by spam clicking the button!)

Reply:
The parent function `Dismiss()` is a public though, so multiple instances of this coroutine could still be created at once. I think it'd be good to but in safeguards against any unexpected behaviour from that side of things.

Reply:
Alrighty, that's fair.

Reply:
Locking thingy added.